ci/build_wheel.sh

```diff
@@ -49,7 +49,7 @@ fi
 
 cd ${package_dir}
 
-SKBUILD_CONFIGURE_OPTIONS="-DDETECT_CONDA_ENV=OFF -DDISABLE_DEPRECATION_WARNINGS=ON -DCPM_cumlprims_mg_SOURCE=${GITHUB_WORKSPACE}/cumlprims_mg/" \
+SKBUILD_CMAKE_ARGS="-DDETECT_CONDA_ENV=OFF;-DDISABLE_DEPRECATION_WARNINGS=ON;-DCPM_cumlprims_mg_SOURCE=${GITHUB_WORKSPACE}/cumlprims_mg/" \
```

Comment:
Just noting in case you haven't seen it... `pip install` and `pip wheel` can be instructed to pass CMake defines like this down to `scikit-build-core` via `--config-settings`, like this:

```shell
python -m pip wheel \
    -w dist \
    -vvv \
    --no-deps \
    --disable-pip-version-check \
    --config-settings=cmake.define.DETECT_CONDA_ENV=OFF \
    --config-settings=cmake.define.DISABLE_DEPRECATION_WARNINGS=ON \
    --config-settings=cmake.define.CPM_cumlprims_mg_SOURCE="${GITHUB_WORKSPACE}/cumlprims_mg/" \
    .
```

Doing that instead of a shell variable means you don't have to worry about spaces vs. semicolons, or about trying to read a very long line if more arguments get added into this string.

`build` also supports this mechanism (although it's called `config-setting` without an `s` there).

That's described at https://scikit-build-core.readthedocs.io/en/latest/configuration.html#configuring-cmake-arguments-and-defines.

I'm not familiar enough with `cuml`'s build process to know if there are other things that make the `SKBUILD_CMAKE_ARGS` environment variable preferable to `--config-settings` here, so do whatever you want with this comment. Just saw this PR go by and thought I should mention it.

Reply:
Yup thanks, I am familiar with that support. There are definitely tradeoffs, but the main reasons I preferred to keep it this way are 1) consistency with the old approach, and 2) (more relevant for other packages) the ability to define the env var in one script and then run another script that does the build. (2) is relevant for cases like raft where `build_wheel_pylibraft.sh` or `build_wheel_raft_dask.sh` sets the vars then calls `build_wheel.sh` (see rapidsai/raft#2051).

Reply:
makes sense to me, thanks for that explanation